hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcServer.java

```diff
@@ -428,6 +456,25 @@ public RouterRpcServer(Configuration conf, Router router,
     initRouterFedRename();
   }
 
+  private void initAsyncThreadPool() {
+    int asyncHandlerCount = conf.getInt(DFS_ROUTER_RPC_ASYNC_HANDLER_COUNT,
+        DFS_ROUTER_RPC_ASYNC_HANDLER_COUNT_DEFAULT);
+    int asyncResponderCount = conf.getInt(DFS_ROUTER_RPC_ASYNC_RESPONDER_COUNT,
+        DFS_ROUTER_RPC_ASYNC_RESPONDER_COUNT_DEFAULT);
+    synchronized (RouterRpcServer.class) {
+      if (asyncRouterHandler == null) {
+        LOG.info("init router async handler count: {}", asyncHandlerCount);
+        asyncRouterHandler = Executors.newFixedThreadPool(
+            asyncHandlerCount, new AsyncThreadFactory("router async handler "));
+      }
+      if (asyncRouterResponder == null) {
+        LOG.info("init router async responder count: {}", asyncResponderCount);
+        asyncRouterResponder = Executors.newFixedThreadPool(
+            asyncHandlerCount, new AsyncThreadFactory("router async responder "));
```

Comment:
@KeeProMise Hi, sir. here should be asyncResponderCount right?

Reply:
@hfutatzhanghb Yes, I will change this to asyncResponderCount later. Thank you for your careful review.